platform/src/main/java/org/stellar/anchor/platform/PaymentConfig.java

```diff
@@ -58,16 +63,30 @@ public StellarPaymentObserver stellarPaymentObserverService(
             .horizonServer(appConfig.getHorizonUrl())
             .paymentTokenStore(stellarPaymentStreamerCursorStore)
             .observers(paymentListeners)
-            .accounts(
-                stellarAssets.stream()
-                    .map(AssetInfo::getDistributionAccount)
-                    .collect(Collectors.toList()))
+            .paymentObservingAccountManager(paymentObservingAccountsManager)
             .build();
 
+    for (AssetInfo assetInfo : stellarAssets) {
+      if (!paymentObservingAccountsManager.observe(assetInfo.getDistributionAccount())) {
+        paymentObservingAccountsManager.upsert(assetInfo.getDistributionAccount(), false);
+      }
+    }
+
     stellarPaymentObserverService.start();
     return stellarPaymentObserverService;
   }
 
+  @Bean
+  public PaymentObservingAccountsManager observingAccounts(
+      PaymentObservingAccountStore paymentObservingAccountStore) {
+    return new PaymentObservingAccountsManager(paymentObservingAccountStore);
+  }
+
+  @Bean
+  public PaymentObservingAccountStore observingAccountStore(PaymentObservingAccountRepo repo) {
```

Comment:
When a bean is initialized and is passed an interface for an arg, how is this interface initialized?

Reply:
The concrete class of `interface PaymentObservingAccountRepo` is generated by Spring and added to the Spring IoC container. This line requests an object of type `interface PaymentObservingAccountRepo`. Spring then searches the container, finds the class that implements the interface, and pass the object to the `@Bean` annotated function.

Reply:
Thanks for explaining!